### DISCUSSION TITLE
Bot Framework Composer DirectLine

### DISCUSSION THREAD

**Discussion by projalm** [microsoft/BotFramework-Composer#9341] [Q&A] [answered]
I hope you can help me in this problem I have created a chatbot in the azure framework Bot Framework Composer and I want to use it in WhatsApp Bussines . My problem comes for the reason that bot framework composer It still does not have a direct channel to whatsapp or any platform like Twilio or Gupshup. I was recommended to use the azure "DirectLine" channel that comes integrated, to take the endpoint of my chatbot produced by azure when deploying it in my group of resources and so with the keys of the channel that I provides "DirecLine" to be able to interact with my chatbot. try do the tests with the endpoint of my chatbot through Postman but I have not been successful, despite following the documentation as indicated microsoft, gives me an answer that is not authorized

I follow this steps

![Captura de pantalla (27)](https://user-images.githubusercontent.com/85646291/183678859-212e10f9-a591-45d0-a25d-37398223efa1.png)
![Captura de pantalla (28)](https://user-images.githubusercontent.com/85646291/183678888-09806d03-e3db-45ee-9bd8-62a0fa02a967.png)
![Captura de pantalla (31)](https://user-images.githubusercontent.com/85646291/183678907-d60990ad-1bff-4839-add2-9adae6f1a15f.png)
![Captura de pantalla (29)](https://user-images.githubusercontent.com/85646291/183678945-7bc0a3e3-2894-4683-aca3-ea5c058d78e6.png)
![Captura de pantalla (30)](https://user-images.githubusercontent.com/85646291/183678960-138f7ac4-d64d-431d-8c1f-1c9d7f8bd47a.png)

**Comment by rschuetzler** [comment, marked as answer]
My understanding of how to get it to work: 

1. Get your key from the Site for Direct Line, not the App Service Extension (it looks in yours like you need to create a site by clicking "Nuevo Sito")
2. Create a conversation by sending a request like the one below. Notice the request goes to `https://directline.botframework.com/v3/directline/conversations/`, not the site's URL. The bearer token should be the site secret keys you got in (1): 
![Screenshot 2022-12-01 at 3 06 18 PM](https://user-images.githubusercontent.com/600775/205169140-74421c6c-212a-4122-8447-68c453955a21.png)
3. Send a message like the one you had, but to `https://directline.botframework.com/v3/directline/conversations/{{conversation_id}}/activities`, where the {{conversation_id}} comes from the response to (2).
